### PR TITLE
Feat: Bundle size action

### PR DESCRIPTION
**Pull request by jackw**
This PR introduces a new action to surface frontend bundle size changes. It relies on the Github actions asset storage to track stats on the default branch. PRs will trigger stats builds and the actions script then compares the changes to asset and module sizes and if the total entrypoint difference is above the minimum threshold the action will post a comment on the PR with a breakdown. Various example comments using this action can be [found here](https://github.com/jackw/heywesty-trafficlight-panel/pull/46).

A standard comment where the size change of entrypoints is greater than the threshold can be [seen here](https://github.com/jackw/heywesty-trafficlight-panel/pull/46#issuecomment-2426975374).

image

A similar comment that reduces size of entrypoints can be [seen here](https://github.com/jackw/heywesty-trafficlight-panel/pull/46#issuecomment-2469772440).

image

A comment where a PR was previously above the minimum threshold but later falls within it can be [seen here](https://github.com/jackw/heywesty-trafficlight-panel/pull/46#issuecomment-2470083443). I'm open to the idea of deleting it entirely as well but thought this more informative. 🤷 

image

### CODE REVIEW

File: bundle-size/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+## Troubleshooting
+
+**How do I add the `main-branch-stats` asset to my github repo?**
+
+To add and update `main-branch-stats` in your github repo you can make the following changes to your workflow that runs on merges to your default (`main`) branch:
+
+1. Replace the frontend build command so on pushes to main the stats.json file is generated by webpack:
+
+```diff
```
Comment (academo) [inline bundle-size/README.md:57]: which file is this? maybe let's use the one we use in our create-plugin template as example?

File: bundle-size/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      ...repo,
+      issue_number: prNumber,
+    });
+
+    const [previousComment, ...restComments] = comments.filter(
+      (comment) => comment.body && comment.body.includes(prMessageSymbol)
+    );
+
+    if (restComments.length > 1) {
+      console.log("Cleaning up old comments... 🧹");
```
Comment (academo) [inline bundle-size/index.js:41]: are you certain we are removing "old" comments and not keeping the oldest and removing the newer ones?

Doesn't really matter for this case, but making sure there's no important reasoning behind updating only the latest

File: bundle-size/index.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            entriesDiff.total.diffPercentage,
+            diffThreshold
+          ),
+        });
+      }
+      console.log(`${msg}`);
+      core.setOutput(msg);
+      return;
+    }
+
```
Comment (academo) [inline bundle-size/index.js:70]: nit. only create `const commentBody = getComment(assetsDiff, modulesDiff, entriesDiff);` after this line or even put the return value directly to `body`

**Review by academo — APPROVED** [review]

File: bundle-size/README.md (hunk 1 above)
Comment (jackw) [reply bundle-size/README.md:57]: Have updated the readme to mention file paths with the assumption that we're using create-plugin workflows.

**Pull request merged**